sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapExamples.cs

```diff
@@ -123,5 +123,124 @@ public void InterestRateSwapPortfolioCashFlowsExample(ModelSelection.ModelEnum m
+            TestDataUtilities.CheckNoCashPositionsInValuationResults(
+                valuationBeforeAndAfterExpirationInterestRateSwap,
+                currency);
+            TestDataUtilities.CheckNonZeroPvBeforeMaturityAndZeroAfter(
+                valuationBeforeAndAfterExpirationInterestRateSwap,
+                interestRateSwap.MaturityDate);
+
+            // GET all upsertable cashflows (transactions) for the InterestRateSwap.
+            // EffectiveAt after maturity so we have all the data.
+            var effectiveAt = interestRateSwap.MaturityDate.AddDays(1);
+            var allInterestRateSwapCashFlows = _transactionPortfoliosApi.GetUpsertablePortfolioCashFlows(
+                    scope,
+                    portfolioCode,
+                    effectiveAt,
+                    windowStart,
+                    windowEnd,
+                    null,
+                    null,
+                    scope,
+                    recipeCode)
+                .Values;
+
+            // Check that some cash flows were returned
+            Assert.That(allInterestRateSwapCashFlows, Is.Not.Empty);
+
+            // For the portfolio to contain the cash flows coming from the InterestRateSwap
+            // we have to upsert the transactions we obtained from the GetUpsertablePortfolioCashFlows endpoint above.
+            // First populate the cashflow transactions with unique IDs.
+            var upsertCashFlowTransactions = tutorials.Ibor.PortfolioCashFlows.PopulateCashFlowTransactionWithUniqueIds(
+                allInterestRateSwapCashFlows,
+                currency);
+
+            // Then UPSERT the cash flows back into LUSID.
+            _transactionPortfoliosApi.UpsertTransactions(
+                scope,
+                portfolioCode,
+                tutorials.Ibor.PortfolioCashFlows.MapToCashFlowTransactionRequest(upsertCashFlowTransactions));
+
+            // HAVING upserted both cash flow and underlying into LUSID, we call GetValuation again.
+            var valuationAfterUpsertingCashFlows = _aggregationApi.GetValuation(valuationRequest);
+
+            // ASSERT that we have some cash in the portfolio
+            var luidsContainedInValuationResponse = valuationAfterUpsertingCashFlows
+                .Data
+                .Select(dictionaryOfMetrics => (string)dictionaryOfMetrics[TestDataUtilities.Luid]);
+            Assert.That(luidsContainedInValuationResponse, Does.Contain($"CCY_{currency}"));
+
+
+            // Everytime a cash flow goes transaction occurs, the PV changes accordingly in an equal and opposite amount.
+            // Thus we expect the PV to be constant throughout.
+
+            // Note all the dates on which cash flows occur.
+            var cashFlowDates = allInterestRateSwapCashFlows
+                .Select(cashFlowTransaction => cashFlowTransaction.TransactionDate)
+                .Distinct()
+                .ToList();
+
+            foreach (var cashFlowDate in cashFlowDates)
```

Comment:
optional: add a small comment above this foreach loop to explain

Reply:
added "CHECK that the PV of the portfolio is constant across dates that are in a small neighbourhood of every cash flow transaction date"